File: src/main/java/com/example/work/entity/UserRole.java
package com.example.work.entity;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.authority.SimpleGrantedAuthority;

import java.util.Set;
import java.util.stream.Collectors;

@Getter
@RequiredArgsConstructor
public enum UserRole {
    TRAINING_REPRESENTATIVE(Set.of(Permission.DEVELOPERS_READ))
    , LECTURER(Set.of(Permission.DEVELOPERS_READ, Permission.DEVELOPERS_WRITE))
    , STUDENT(Set.of(Permission.DEVELOPERS_READ));

    private final Set<Permission> permissions;

    public Set<SimpleGrantedAuthority> getAuthorities() {
        return permissions.stream()
                .map(permission -> new SimpleGrantedAuthority(permission.getPermission()))
                .collect(Collectors.toSet());
    }
}

File: src/main/java/com/example/work/model/UserRole.kt
package com.example.work.model

import org.springframework.security.core.authority.SimpleGrantedAuthority

enum class UserRole(private val permissions: Set<Permission>) {
    TRAINING_REPRESENTATIVE(setOf(Permission.DEVELOPERS_READ)),
    LECTURER(setOf(
            Permission.DEVELOPERS_READ,
            Permission.DEVELOPERS_WRITE,
            Permission.COURSES_READ,
    )),
    STUDENT(setOf(
            Permission.DEVELOPERS_READ,
    ));

    val authorities: Set<SimpleGrantedAuthority>
        get() = permissions
            .map { SimpleGrantedAuthority(it.permission) }
            .toSet()
}
